### ISSUE TITLE
Upgrading to php 8.0 and trying to run update.php

### ISSUE THREAD

**Issue by plegault3397** [SemanticMediaWiki/SemanticMediaWiki#5304]
### Setup

- SMW version: 4.0.2
- MW version: 1.35.8
- PHP version: 8.0.22 
- DB system (MySQL, Blazegraph, etc.) and version: 10.4.25-MariaDB

### Issue

Detailed description of the issue and a [stack trace](https://www.semantic-mediawiki.org/wiki/Help:Identifying_bugs) if applicable:

```
Trying to run update.php
[71f2eca0ccb973e1a2cdcd25] [no req]   TypeError from line 32 of D:\data\mediawiki\vendor\onoi\http-request\src\CurlRequest.php: get_resource_type(): Argument #1 ($resource) must be of type resource, CurlHandle given
Backtrace:
#0 D:\data\mediawiki\vendor\onoi\http-request\src\CurlRequest.php(32): get_resource_type()
#1 D:\data\mediawiki\extensions\SemanticMediaWiki\src\SPARQLStore\RepositoryConnectionProvider.php(177): Onoi\HttpRequest\CurlRequest->__construct()
#2 D:\data\mediawiki\extensions\SemanticMediaWiki\src\SPARQLStore\RepositoryConnectionProvider.php(159): SMW\SPARQLStore\RepositoryConnectionProvider->connectTo()
#3 D:\data\mediawiki\extensions\SemanticMediaWiki\src\Connection\ConnectionManager.php(46): SMW\SPARQLStore\RepositoryConnectionProvider->getConnection()
#4 D:\data\mediawiki\extensions\SemanticMediaWiki\src\Store.php(581): SMW\Connection\ConnectionManager->getConnection()
#5 D:\data\mediawiki\extensions\SemanticMediaWiki\src\SPARQLStore\SPARQLStore.php(512): SMW\Store->getConnection()
#6 D:\data\mediawiki\extensions\SemanticMediaWiki\src\SPARQLStore\SPARQLStore.php(387): SMW\SPARQLStore\SPARQLStore->getConnection()
#7 D:\data\mediawiki\extensions\SemanticMediaWiki\src\Store.php(489): SMW\SPARQLStore\SPARQLStore->setup()
#8 D:\data\mediawiki\includes\installer\DatabaseUpdater.php(554): SMW\Store::setupStore()
#9 D:\data\mediawiki\includes\installer\DatabaseUpdater.php(522): DatabaseUpdater->runUpdates()
#10 D:\data\mediawiki\maintenance\update.php(181): DatabaseUpdater->doUpdates()
#11 D:\data\mediawiki\maintenance\doMaintenance.php(107): UpdateMediaWiki->execute()
#12 D:\data\mediawiki\maintenance\update.php(253): require_once(string)
#13 {main}

PHP 7.3 is End of Life and 7.4 will be soon: 
```
[PHP Support](https://www.php.net/supported-versions.php)

Steps to reproduce the observation (recommendation is to use the [sandbox](https://sandbox.semantic-mediawiki.org)):
run update.php

**Comment by malberts** [comment]
Can you try again with the latest master branch?

I believe this is fixed as part of #5351. It is the same underlying error, although it was failing in tests, not during update.php.

**Comment by Seb35** [comment]
> Can you try again with the latest master branch?
> 
> I believe this is fixed as part of #5351. It is the same underlying error, although it was failing in tests, not during update.php.

No answer by the reporter @plegault3397 but this is almost a duplicate of #5346 which is fixed by @malberts in #5351.
